packages/neo4j-driver/test/internal/timers-util.js

```diff
@@ -15,47 +15,37 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
+*/
+
+/**
+ * This is a liter mock which only creates mocked functions to work
+ * as timeouts.
  */
-class SetTimeoutMock {
+class TimeoutsMock {
   constructor () {
-    this._clearState()
+    this.clearState()
+    // bind it to be used as standalone functions
+    this.setTimeout = this.setTimeout.bind(this)
+    this.clearTimeout = this.clearTimeout.bind(this)
   }
 
-  install () {
-    this._originalSetTimeout = global.setTimeout
-    global.setTimeout = (code, delay) => {
-      if (!this._paused) {
-        code()
-        this.invocationDelays.push(delay)
-      }
-      return this._timeoutIdCounter++
-    }
-
-    this._originalClearTimeout = global.clearTimeout
-    global.clearTimeout = id => {
-      this.clearedTimeouts.push(id)
+  setTimeout (code, delay) {
+    if (!this._paused) {
+      code()
+      this.invocationDelays.push(delay)
     }
+    return this._timeoutIdCounter++
```

Comment:
```suggestion
    let timeoutId = this._timeoutIdCounter++
    this.invocationDelays.push(delay)
    if (!this._paused) {
      code()
    }
    return timeoutId
```

I'm not sure whether there's a reason not to do it like this, but this would've been my first intuition: record the call (including the delay) regardless if `code()` throws or weather callbacks are `_paused`.